dcrsqlite/apisource.go

```diff
@@ -468,3 +468,104 @@ func (db *wiredDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails {
 	}
 	return &mpTicketDetails
 }
+
+//GetAddressTransactions returns an apitypes.Address Object
+//with at most the last 10 transactions the address was in
+func (db *wiredDB) GetAddressTransactions(addr string, count int) *apitypes.Address {
+	address, err := dcrutil.DecodeAddress(addr)
+	if err != nil {
+		log.Errorf("Invalid address %s", addr)
+		return nil
+	}
+	txs, err := db.client.SearchRawTransactionsVerbose(address, 0, count, true, false, make([]string, 0))
+	if err != nil {
+		log.Errorf("GetAddressTransactions failed for address %s", addr)
+		return nil
+	}
+	tx := make([]*apitypes.AddressTxShort, 0)
```

Comment:
If making a zero-length slice, it is more idiomatic to do:

```go
tx := []*apitypes.AddressTxShort{} // empty slice
// or
var tx []*apitypes.AddressTxShort // nil slice
```

These are technically different, but it rarely matters which you use.

However, if you are appending in a loop of known length, you should create a zero-length slice with the right capacity (for performance reasons):

```go
txarray := make([]*apitypes.AddressTxShort, 0, len(txs))
```